SQL assignment.txt

```diff
@@ -0,0 +1,1533 @@
+| cnum | cname    | city    | rating | snum |
++------+----------+---------+--------+------+
+| 2001 | hoffman  | london  |    100 | 1001 |
+| 2002 | giovamni | rome    |    200 | 1003 |
+| 2003 | liu      | sanjose |    200 | 1002 |
+| 2004 | grass    | berlin  |    300 | 1002 |
+| 2006 | cisneras | london  |    100 | 1001 |
+| 2006 | cisneras | sanjose |    300 | 1007 |
+| 2007 | pereira  | rome    |    100 | 1004 |
++------+----------+---------+--------+------+
+
+
+
+10) Match salespeople to customers according to what city they live in.
+
+mysql>  SELECT salespeople.* FROM salespeople LEFT JOIN customers ON customers.city = salespeople.city;
+
++------+---------+-----------+------+
+| snum | sname   | city      | comm |
++------+---------+-----------+------+
+| 1001 | Peel    | London    |   12 |
+| 1001 | Peel    | London    |   12 |
+| 1002 | serres  | sanjose   |   13 |
+| 1002 | serres  | sanjose   |   13 |
+| 1003 | axeirod | new york  |   10 |
+| 1004 | motika  | london    |   11 |
+| 1004 | motika  | london    |   11 |
+| 1007 | rifkin  | barcelona |   15 |
+| 1008 | fran    | london    |   25 |
+| 1008 | fran    | london    |   25 |
++------+---------+-----------+------+
+
+
+
+11) Find all the customers in SanJose who have a rating above 200. 
+
+mysql> select * from customers WHERE city="sanjose" and rating > 200;
+
++------+----------+---------+--------+------+
+| cnum | cname    | city    | rating | snum |
++------+----------+---------+--------+------+
+| 2006 | cisneras | sanjose |    300 | 1007 |
++------+----------+---------+--------+------+
+
+
+
+12) List the names and commissions of all salespeople in London.
+
+mysql> select * from salespeople where city="london";
+
++------+--------+--------+------+
+| snum | sname  | city   | comm |
++------+--------+--------+------+
+| 1001 | Peel   | London |   12 |
+| 1004 | motika | london |   11 |
+| 1008 | fran   | london |   25 |
++------+--------+--------+------+
+
+
+
+13) List all the orders of Salesperson Motika from the orders table.
+
+mysql> select *,snum from orders where snum = 1004;
+
++------+--------+----------+------+------+------+
+| onum | amt    | odate    | cnum | snum | snum |
++------+--------+----------+------+------+------+
+| 3002 | 1900.1 | 10/03/90 | 2007 | 1004 | 1004 |
++------+--------+----------+------+------+------+
+
+
+
+14) Find all customers who booked orders on October 3.
+
+mysql> select distinct c.cname,c.cnum from customers as c join orders as o on c.cnum = o.cnum where o.odate = '10/03/90';
+
++---------+------+
+| cname   | cnum |
++---------+------+
+| hoffman | 2001 |
+| liu     | 2003 |
+| pereira | 2007 |
++---------+------+
+
+
+
+15) Give the sums of the amounts from the Orders table, grouped by date, eliminating all those dates where the SUM was not at least 2000 above the maximum Amount.
+
+mysql> SELECT odate, SUM(amt) AS total_for_date FROM orders GROUP BY odate  HAVING total_for_date - 2000 < ( SELECT MAX(amt) FROM orders);
+
```

Comment:
Incorrect query. Use ">=" instead of "<".

Reply:
resolved